### PR TITLE
Add primitive warnings

### PR DESCRIPTION
**Pull request by martijnbastiaan**
Original PR: https://github.com/clash-lang/clash-compiler/pull/286. I've rebased it on top of master and will start implementing @basile-henry 's suggestions.

- [x] Print warning to stderr instead of stdout
- [x] Coloured warnings (orange?). EDIT: it prints bold/underline text now. I'm not really sure how to properly change it to colors..
- [x] Disable warnings for the testbench netlist generation
- [x] Remove duplicate warnings (for the same primitive)

### CODE REVIEW

File: clash-ghc/src-ghc/Clash/GHC/ClashFlags.hs [hunk 1]
```diff
@@ -60,6 +60,8 @@ flagsClash r = [
   , defFlag "fclash-hdlsyn"                  $ SepArg (setHdlSyn r)
   , defFlag "fclash-nocache"                 $ NoArg (liftEwM (setNoCache r))
   , defFlag "fclash-noclean"                 $ NoArg (liftEwM (setNoClean r))
+  , defFlag "fclash-no-prim-warn"            $ NoArg (liftEwM (setNoPrimWarn r))
+  , defFlag "fclash-no-color"                $ NoArg (liftEwM (setNoColor r))
```
Comment (leonschoorl) [inline clash-ghc/src-ghc/Clash/GHC/ClashFlags.hs]: Would be cool if could hook into GHC's [-fdiagnostics-color](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/flags.html#ghc-flag--fdiagnostics-color=%E2%9F%A8always|auto|never%E2%9F%A9)

Comment (leonschoorl) [reply clash-ghc/src-ghc/Clash/GHC/ClashFlags.hs]: Also we should rename the the others to `-fclash-no-cache` and `-fclash-no-clean` to be consistent with the new flags and with all of [GHC's flags](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/flags.html).

Comment (martijnbastiaan) [reply clash-ghc/src-ghc/Clash/GHC/ClashFlags.hs]: Agreed to both. I'll create a separate issue for the flag renaming, as I think it's too invasive to handle in this PR. I'll look into `-fdiagnostic-color`, thanks!

**Comment by leonschoorl** [comment]
I meant the opposite direction: remove the `-fclash-no-color` flag and have `-fdiagnostics-color` control whether or not the clash error messages use colors.

**Comment by martijnbastiaan** [comment]
Do we "intercept" other GHC flags this way? If not, I'm not sure if it's a good precedence to set. I don't have any good arguments against it though, except that "it feels inconsistent".

**Comment by martijnbastiaan** [comment]
Probably. I'll check it out. If it's not feasible I'll just keep the current approach. If it is, I'll patch it.

**Comment by martijnbastiaan** [comment]
It's non-obvious to me how this would work. I'd like to store whether we use colors in `ClashOpts` (which we pass around later on). We generate a `ClashOpts` upon parsing flags and then never change it again AFAIK. The `color` field would have to become a special case, which we change in some GHC monad. 

I don't think this flag warrants such a special case, so I'm going to merge it as is upon no further comments :).

**Comment by martijnbastiaan** [comment]
Actually, Christiaan just pointed out that we only actually 'get' the clashopts from the ioref it's in initially in `makeHDL`. We're already in a GHCMonad there, so we could just change the opts there.

**Pull request merged**